addons/website_event_track/static/src/js/website_event_track_proposal.js

```diff
@@ -0,0 +1,185 @@
+            });
+            
+            $field.find('.form-control').removeClass('o_has_error').removeClass('is-invalid');
+            if (invalid_inputs.length) {
+                $field.find('.form-control').addClass('o_has_error').addClass('is-invalid');
+                formErrors.push('invalidFormInputs');
+            }
+        });
+
+        // 2) Advanced Contact Must Have a Contact Mean
+        if (this.useAdvancedContact) {
+            var hasContactMean = this.$('.o_wetrack_contact_phone_input').val() || 
+                                 this.$('.o_wetrack_contact_email_input').val();
+            if (!hasContactMean) { 
+                this.$('.o_wetrack_contact_information').addClass('o_has_contact_mean_error'); 
+                this.$('.o_wetrack_contact_mean').addClass('is-invalid');
+                formErrors.push('noContactMean');
+            } else {
+                this.$('.o_wetrack_contact_information').removeClass('o_has_contact_mean_error'); 
+                this.$('.o_wetrack_contact_mean:not(".o_has_error")').removeClass('is-invalid');
+            }
+        }
+
+        // Form Validity and Error Display
+        this.isFormValid = formErrors.length < 1;
+        this._updateErrorDisplay(formErrors);
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Display / Hide Additional Contact Information section when toggling
+     * the checkbox on the form o_wetrack_add_contact_information_checkbox.
+     * Also empty the email to prevent hidden email format error.
+     * 
+     * @private
+     * @param {Event} ev
+     */
+    _onAdvancedContactToggle: function(ev) {
+        this.useAdvancedContact = !this.useAdvancedContact;
+        var $advancedInformation = this.$('.o_wetrack_contact_information');
+
+        if (this.useAdvancedContact) {
+            $advancedInformation.removeClass('d-none');
+        } else {
+            this.$('.o_wetrack_contact_email_input').val('').change();
+            $advancedInformation.addClass('d-none');
+        }
+    },
+
+    /**
+     * Propagates the speaker name to contact name on new input. 
+     * 
+     * @private
+     * @param {Event} ev
+     */
+    _onNameInputPropagate: function(ev) {
+        var text = $(ev.currentTarget).val();
+        this.$(".o_wetrack_contact_name_input").val(text).change();
+    },
+
+    /**
+     * Submits the form if no errors are present in the form after validation.
+     * 
+     * @private
+     * @param {Event} ev
+     */
+    _onSubmitProposalForm: function(ev) {
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        // Prevent further clicking
+        this.$target.find('.o_wetrack_proposal_submit_button')
+            .addClass('disabled')
+            .attr('disabled', 'disabled');
+
+        this._validateForm();
```

Comment:
Could change to _isFormValid() that returns the isValid status instead of playing with widget properties.

Reply:
Done, indeed.